seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/task/SeaTunnelSourceCollector.java

```diff
@@ -100,21 +119,33 @@ public SeaTunnelSourceCollector(
     public void collect(T row) {
         try {
             if (row instanceof SeaTunnelRow) {
+                String tableId = ((SeaTunnelRow) row).getTableId();
                 int size;
                 if (rowType instanceof SeaTunnelRowType) {
                     size = ((SeaTunnelRow) row).getBytesSize((SeaTunnelRowType) rowType);
                 } else if (rowType instanceof MultipleRowType) {
-                    size =
-                            ((SeaTunnelRow) row)
-                                    .getBytesSize(
-                                            rowTypeMap.get(((SeaTunnelRow) row).getTableId()));
+                    size = ((SeaTunnelRow) row).getBytesSize(rowTypeMap.get(tableId));
                 } else {
                     throw new SeaTunnelEngineException(
                             "Unsupported row type: " + rowType.getClass().getName());
                 }
                 sourceReceivedBytes.inc(size);
                 sourceReceivedBytesPerSeconds.markEvent(size);
                 flowControlGate.audit((SeaTunnelRow) row);
+                if (StringUtils.isNotEmpty(tableId)) {
+                    String tableName = TablePath.of(tableId).getTableName();
+                    if (StringUtils.isNotBlank(tableName)) {
+                        Counter sourceTableCounter = sourceReceivedCountPerTable.get(tableName);
+                        if (Objects.nonNull(sourceTableCounter)) {
+                            sourceTableCounter.inc();
+                        } else {
+                            sourceReceivedCountPerTable.put(
+                                    tableName,
+                                    metricsContext.counter(
+                                            SOURCE_RECEIVED_COUNT + "#" + tableName));
```

Comment:
only put but not invoke `inc`? Seem like it will lose 1 row.

Reply:
Fixed